src/components/clusterDetail/ClusterInstallationProgressCard.tsx

```diff
@@ -0,0 +1,119 @@
+  const [credentialsError, setCredentialsError] = React.useState();
+  const [isCardExpanded, setIsCardExpanded] = React.useState(cluster.status !== 'installed');
+
+  const fetchCredentials = React.useCallback(() => {
+    const fetch = async () => {
+      setCredentialsError(undefined);
+      try {
+        const response = await getClusterCredentials(cluster.id);
+        setCredentials(response.data);
+      } catch (err) {
+        setCredentialsError(err);
+      }
+    };
+    fetch();
+  }, [cluster.id]);
+
+  React.useEffect(() => {
+    if (cluster.status === 'installed') {
+      fetchCredentials();
+    }
+  }, [cluster.status, fetchCredentials]);
+
+  return (
+    <Card id="ai-cluster-details-card" isExpanded={isCardExpanded}>
+      <AlertsContextProvider>
+        <CancelInstallationModal
+          isOpen={cancelInstallationModalOpen}
+          onClose={() => setCancelInstallationModalOpen(false)}
+          clusterId={cluster.id}
+        />
+        <ResetClusterModal
+          isOpen={resetClusterModalOpen}
+          onClose={() => setResetClusterModalOpen(false)}
+          cluster={cluster}
+        />
+      </AlertsContextProvider>
+      <CardHeader
+        onExpand={() => setIsCardExpanded(!isCardExpanded)}
+        toggleButtonProps={{
+          id: 'toggle-ai-cluster-details',
+          'aria-label': 'AI Cluster Details',
+          'aria-labelledby': 'titleId toggle-button',
+          'aria-expanded': isCardExpanded,
+        }}
+      >
+        <Title headingLevel="h1" size="lg" className="card-title">
+          Installation progress
+        </Title>
+        {!isCardExpanded && (
+          <div className="pf-u-ml-md">
+            <ClusterStatusIcon status={cluster.status} />
+          </div>
+        )}
+      </CardHeader>
+      <CardBody>
+        <ClusterProgress cluster={cluster} minimizedView={!isCardExpanded} />
+        <ClusterDetailsButtonGroup cluster={cluster} />
+      </CardBody>
+      <CardExpandableContent>
+        <CardBody>
+          {['installed', 'installing', 'finalizing'].includes(cluster.status) && (
+            <FailedHostsWarning cluster={cluster} />
+          )}
+          {cluster.status === 'error' && (
+            <ClusterInstallationError
+              cluster={cluster}
+              setResetClusterModalOpen={setResetClusterModalOpen}
+            />
+          )}
+          {cluster.status === 'cancelled' && (
+            <ClusterInstallationError
+              title="Cluster installation was cancelled"
+              cluster={cluster}
+              setResetClusterModalOpen={setResetClusterModalOpen}
+            />
+          )}
+          {cluster.status === 'installed' && (
+            <ClusterCredentials
+              cluster={cluster}
+              credentials={credentials}
+              error={!!credentialsError}
+              retry={fetchCredentials}
+              idPrefix={getID('cluster-creds')}
```

Comment:
@mareklibra same thing here for the idPrefix prop. This is going to be translated into some nested element id attr.
If this is used for e2e tests then please stick to `data-testid`